pyperf/_process_time.py

```diff
@@ -125,15 +151,21 @@ def main():
         kw['stdout'] = devnull
     kw['stderr'] = subprocess.STDOUT
 
-    dt, max_rss = bench_process(loops, args, kw, profile_filename)
+    with contextlib.ExitStack() as stack:
+        for hook in hook_managers.values():
+            stack.enter_context(hook)
+        dt, max_rss = bench_process(loops, args, kw, profile_filename)
 
     if devnull is not None:
         devnull.close()
 
+    for hook in hook_managers.values():
+        hook.teardown(metadata)
+
     # Write timing in seconds into stdout
     print(dt)
```

Comment:
@mdboom Would you like to extract printing data logic into a single function and then add a comment about the function from the parsing metadata logic at bench_command to explain why we parse that way?